svgdigitizer/electrochemistry/cv.py

```diff
@@ -19,110 +19,199 @@
+        ...   </g>
+        ...   <g>
+        ...     <path d="M -100 0 L 0 0" />
+        ...     <text x="-100" y="0">y2: 1 A</text>
+        ...   </g>
+        ...   <text x="-200" y="330">scan rate: 50 V/s</text>
+        ... </svg>'''))
+        >>> cv = CV(SVGPlot(svg))
+        >>> cv.rate
+        <Quantity 50. V / s>
 
-    def modify_df(self):
-        # Create potential columns
-        self.df = self.create_df_U_axis(self.svgplot.dfs[0])
-        # Create current columns
-        self.df = pd.concat(
-            [self.df, self.create_df_I_axis(self.svgplot.dfs[0])],
-            axis=1)
+        """
+        rates = self.svgplot.svg.get_texts('(?:scan rate|rate): (?P<value>-?[0-9.]+) *(?P<unit>.*)')
+        # To Do: assert that only one label contains the scan rate (see issue #58)
+        # To Do: assert that a rate is available at all (see issue #58)
 
-        # create time axis
-        self.df['t'] = self.create_df_time_axis(self.svgplot.dfs[0])
+        # Convert to astropy unit
+        rates[0].unit = CV.get_axis_unit(rates[0].unit)
+        rate = float(rates[0].value) * rates[0].unit
 
-    def create_df_U_axis(self, df):
-        r'''
-        Create voltage axis in the dataframe based on the units given in the
-        figure description.
-        '''
-        df_ = df.copy()
-        # Call a dict and remove the if functions
-        if self.xunit == 'V':
-            df_['U_V'] = df['U']
+        return rate
 
-        if self.xunit == 'mV':
-            df_['U_V'] = df['U']/1E3
+    @property
+    @cache
+    def df(self):
+        r"""
+        Return a dataframe with axis 't', 'U', and 'I' (or 'j).
+        The dimensions are in SI units 's', 'V' and 'A' (or 'A / m2').
 
-        df_['U_mV'] = df_['U_V']*1E3
+        The dataframe is constructed from the 'x' and 'y' axis of 'svgplot.df',
+        which are usually not in SI units.
+
+        The time axis can only be created when a (scan) rate is given in the plot, i.e., 50 mV /s.
+        """
+        # Add a potential axis.
+        df = self._add_U_axis(self.svgplot.df)
 
-        return df_[['U_V', 'U_mV']]
+        df = self._add_I_axis(self.svgplot.df)
 
-    def create_df_I_axis(self, df):
+        df = self._add_time_axis(df)
+
+        # Rearrange columns.
+        df = df[['t', 'U', self.axis_properties['y']['dimension']]]
+
+        return df
+
+    def _add_U_axis(self, df):
         r'''
-        Create current or current density axis in the dataframe based on the
-        units given in the figure description.
+        Add voltage column to the dataframe `df`, based on the :meth:`get_axis_unit`.
         '''
-        df_ = df.copy()
-        if self.yunit == 'A':
-            df_['I_A'] = df['I']
+        q = 1 * CV.get_axis_unit(self.svgplot.units['x'])
+        # Convert the axis unit to SI unit V and use the value
+        # to convert the potential values in the df to V
+        df['U'] = df['x'] * q.to(u.V).value
+
+        return df
```

Comment:
```suggestion

        return df
```
I would just make all these methods mutation. Currently, that's inconsistent.

Reply:
Same for the other `_add` methods.